Mathlib/Algebra/Module/SpanRank.lean

```diff
@@ -210,6 +220,18 @@ lemma spanRank_sup_le_sum_spanRank {p q : Submodule R M} :
   obtain ⟨sq, ⟨hq₁, rfl⟩⟩ := exists_span_set_card_eq_spanRank q
   exact ⟨sp ∪ sq, ⟨hp₁ ▸ hq₁ ▸ (Cardinal.mk_union_le sp sq), span_union sp sq⟩⟩
 
+lemma spanFinrank_eq_zero_iff_eq_bot {p : Submodule R M} (h : p.FG) :
+    p.spanFinrank = 0 ↔ p = ⊥ := by
+  refine ⟨fun heq ↦ ?_, fun h ↦ h ▸ by simp⟩
+  rw [← Submodule.FG.generators_ncard h, Set.ncard_eq_zero h.finite_generators] at heq
+  rw [← p.span_generators, heq, span_empty]
+
+lemma spanFinrank_singleton {m : M} (hm : m ≠ 0) : (span R {m}).spanFinrank = 1 := by
```

Comment:
Can you deduce this from the more general statement that `spanRank (span R s) = #s` when `s` is linearly independent?

Reply:
Ah yes!